### PR TITLE
Migrating DCOS to Kubernetes post by Rogier

### PR DESCRIPTION
**Pull request by a-nldisr**
Signed-off-by: Rogier Dikkes <1867776+a-nldisr@users.noreply.github.com>

### CODE REVIEW

File: examples/migrating-dcos-kubernetes-l4lb/ipvs-client/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# IPVS client
+
+This application is made fetch ipvs routes from an JSON endpoint, this will be then used to restore these routes on the Kubernetes side. We use it to build an ipvs loadbalancer to make Kubernetes communicate with Mesos/DCOS.
+
+## Requirements
+
+The requirement to get this application running are:
```
Comment (royderks) [inline examples/migrating-dcos-kubernetes-l4lb/ipvs-client/README.md:7]: *requirement**s**

File: examples/migrating-dcos-kubernetes-l4lb/ipvs-client/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+- ipvsadm
+- ipvs kernel modules (in general most Distro's already supply these)
+- sudo / root capabilities.
+
+## Go build
+
+We build our Golang applications with the git sha, if you do not pass this flag then the build should continue.
```
Comment (royderks) [inline examples/migrating-dcos-kubernetes-l4lb/ipvs-client/README.md:15]: Put `git sha` inside code markup?

File: examples/migrating-dcos-kubernetes-l4lb/ipvs-server/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# IPVS server
+
+This application is made to serve `ipvsadm -L -n` output in a json endpoint, this can then be picked up by a client to ingest this. We use it to build an ipvs loadbalancer to make Kubernetes communicate with Mesos/DCOS.
+
+## Requirements
+
+The requirement to get this application running are:
```
Comment (royderks) [inline examples/migrating-dcos-kubernetes-l4lb/ipvs-server/README.md:7]: Same as for the client

File: examples/migrating-dcos-kubernetes-l4lb/ipvs-server/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+You can build the Dockerfile with the following command:
+
+```bash
+docker build --build-arg=GIT_COMMIT="$(git rev-list -1 HEAD)" -f "${DOCKER_FILE_LOC}" -t "${DOCKER_REPO}/${TEAM}/${APP_NAME}:${VERSION}" .
+```
+
+## Go build
+
+We build our Golang applications with the git sha, if you do not pass this flag then the build should continue.
```
Comment (royderks) [inline examples/migrating-dcos-kubernetes-l4lb/ipvs-server/README.md:23]: Same as for the client

File: public/posts/migrating-dcos-kubernetes-l4lb.md [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+<https://github.com/Vandebron/tech-blog>
+
+First, let’s discuss the library we use: <https://github.com/nanobox-io/golang-lvs>. This library in its essence translates to ipvsadm commands, it helped save us time to implement this ourselves. There are some gotcha’s, such as newlines are not filtered out from the output. We solved this by cleaning up some of the data.
+
+In the `childChan` function we create a go channel that is responsible for polling `ipvsadm` every 10 seconds and stores the result in a couple of variables we use in our HTTP endpoints. IPVS is a Linux kernel functionality and should be highly performant, we do not want to trigger kernel panics when the server gets overloaded with requests. We expect that every 10 seconds gives us accurate enough results, we can always lower this interval to ensure faster results. We also added in this function the string manipulation to ensure all the newlines were gone in the JSON output. The newline gave issues when we tried to add the ipvs scheduler entries.
```
[the remark this answers is not in the document]
Comment (a-nldisr) [reply public/posts/migrating-dcos-kubernetes-l4lb.md:69]: IPVS, good one

File: public/posts/migrating-dcos-kubernetes-l4lb.md [hunk 6]
```diff
@@ -0,0 +1,393 @@
...
+
+## L4LB
+
+The internal layer 4 load balancer DCOS provides is used in the majority of our workloads. When our data scientists schedule a spark driver, they connect to the spark dispatcher through the layer4 load balancer. Most of the DCOS frameworks use this layer4 load balancer as an internal service discovery tool, with Vandebron we use this layer 4 load balancer to communicate between services. In a default DCOS set up this load balancer responds on domain names like: spark-dispatcher.marathon.l4lb.thisdcos.directory:7077
+
+When we ping the spark dispatcher we get the following:
+PING spark-dispatcher.marathon.l4lb.thisdcos.directory (11.155.161.35) 56(84) bytes of data.
+64 bytes from 11.155.161.35 (11.155.161.35): icmp_seq=1 ttl=64 time=0.024 ms
+
+After some investigation we found out that this IP range is not actually on a network interface, it is a Linux kernel functionality called ipvs. With ipvs you can do layer 4 load balancing, you provide the target location and the location you want to respond on.
```
[the remark this answers is not in the document]
Comment (a-nldisr) [reply public/posts/migrating-dcos-kubernetes-l4lb.md]: I wrapped the first IPVS in quotes, left the rest as is. I suspect adding all the 30+ IPVS in the text in quotes will make the text a bit unreadable. Let me know if you disagree!

File: public/posts/migrating-dcos-kubernetes-l4lb.md [hunk 7]
```diff
@@ -0,0 +1,393 @@
...
+
+## L4LB
+
+The internal layer 4 load balancer DCOS provides is used in the majority of our workloads. When our data scientists schedule a spark driver, they connect to the spark dispatcher through the layer4 load balancer. Most of the DCOS frameworks use this layer4 load balancer as an internal service discovery tool, with Vandebron we use this layer 4 load balancer to communicate between services. In a default DCOS set up this load balancer responds on domain names like: spark-dispatcher.marathon.l4lb.thisdcos.directory:7077
+
+When we ping the spark dispatcher we get the following:
+PING spark-dispatcher.marathon.l4lb.thisdcos.directory (11.155.161.35) 56(84) bytes of data.
```
[the remark this answers is not in the document]
Comment (a-nldisr) [reply public/posts/migrating-dcos-kubernetes-l4lb.md:20]: Indeed, fixed

File: public/posts/migrating-dcos-kubernetes-l4lb.md [hunk 8]
```diff
@@ -0,0 +1,393 @@
...
+
+## L4LB
+
+The internal layer 4 load balancer DCOS provides is used in the majority of our workloads. When our data scientists schedule a spark driver, they connect to the spark dispatcher through the layer4 load balancer. Most of the DCOS frameworks use this layer4 load balancer as an internal service discovery tool, with Vandebron we use this layer 4 load balancer to communicate between services. In a default DCOS set up this load balancer responds on domain names like: spark-dispatcher.marathon.l4lb.thisdcos.directory:7077
```
[the remark this answers is not in the document]
Comment (a-nldisr) [reply public/posts/migrating-dcos-kubernetes-l4lb.md]: spark-dispatcher.marathon.l4lb.thisdcos.directory:7077 is wrapped.

**Pull request merged**